url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/AgentConnection.java

```diff
@@ -0,0 +1,1042 @@
+			out.println(new InfoPauseDefinition(pauseType).serializeToString());
+		}
+	}
+
+	/**
+	 * Wysyłamy listę trwających połączeń dla nowo zalogowanego agenta
+	 */
+	public void printCallInProgress() {
+		SystemCoreState cs = p.agenci.coreState;
+		synchronized (cs.getQueueList()) {
+			for (RealQueue queue : cs.getQueueList().getValues()){
+				synchronized (queue.getCallList()) {
+					for (RealCall call : queue.getCallList().getValues()){
+						// Połączenie w kolejce
+						//log.warn(">> Queue:" +queue+" call: " +call);
+						String s = "EVENT [Queue], Action [Join], " +
+								"Queue ["+queue.getName()+"], Count ["+queue.getWaitCount()+"], " +
+								"Callerid ["+call.getCallerID()+"], Uniqueid ["+call.getUniqueID()+"], "+
+								"CreationTime ["+call.getCrationTimeFormated()+"]";
+						out.println(s);
+					
+						RealAgent agent = cs.findAgentConnectedWithCall(call);
+						if (agent!=null){
+							//Jeśli połączenie w kolejce sparowane z agentem
+							log.warn(">> connected with: "+agent);
+							// FIXME: Ponieważ klient może jeszcze nie mieć zalogowanego tego agenta to (dostanie to info póżniej)   
+							// EVENT [Agent], Action [Login], Agent [304], Exten [2001], Name [Olsztyńska Antonina]
+							String sAgentLogin = "EVENT [Agent], Action [Login], " +
+									"Agent ["+agent.getId()+"], Exten ["+agent.getExtension()+"], Name ["+agent.getName()+"]";
+							out.println(sAgentLogin);
+							// FIXME:Agenta najprawdopodobniej nie ma także w kolejce 
+							// +INFO Queue 'bluzeczka', Agent '304', Exten '2001', Name: 'Olsztyńska Antonina' (dostanie to info póżniej)
+							String sAgentQueueMember = "+INFO Queue '"+queue.getName()+"', " +
+									"Agent '"+agent.getId()+"', Exten '"+agent.getExtension()+"', Name: '"+agent.getName()+"'";
+							out.println(sAgentQueueMember);
+							
+							//EVENT [Agent], Action [NewCall], Agent [313], Exten [877], Queue [nogawka], CallerID [978]
+							String sNewCall ="EVENT [Agent], Action [NewCall], " +
+									"Agent ["+agent.getId()+"], Exten ["+agent.getExtension()+"], " +
+									"Queue ["+queue.getName()+"], CallerID ["+call.getCallerID()+"], Uniqueid ["+call.getUniqueID()+"],"+
+									"CreationTime ["+call.getCrationTimeFormated()+"]";
+							out.println(sNewCall);
+							//EVENT [Agent], Action [AgentConnect], Agent [304], Exten [2001], Queue [bluzeczka], CallerID [76726470], Holdtime [0], Uniqueid [1246610416.627]
+							String sAgentConnect ="EVENT [Agent], Action [AgentConnect], " +
+									"Agent ["+agent.getId()+"], Exten ["+agent.getExtension()+"], " +
+									"Queue ["+queue.getName()+"], " +
+									"CallerID ["+call.getCallerID()+"], Holdtime [0], Uniqueid ["+call.getUniqueID()+"]," +
+									"CreationTime ["+call.getCrationTimeFormated()+"]";
+							out.println(sAgentConnect);
+						}
+					}
+				}
+			}
+		}
+	}
+	@Override
+	public void run() {
+		runInternal();
+		p.log.info("Thread died");
+	}
+	
+	private void runInternal() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 32. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.